src/vcpkg/commands.install.cpp

```diff
@@ -40,130 +44,202 @@ namespace vcpkg
+        const auto filtered_files = filter_files_to_install(fs, std::move(files));
+
+        auto list_file_future = std::async(std::launch::async | std::launch::deferred, [&]() {
+            install_listfile(fs, prefix_length, destination, destination_dir.listfile(), filtered_files);
+        });
+
+        // Copy directories
+        for (const auto& file : filtered_files)
         {
+            if (!is_directory(file.type)) continue;
+
+            const Path target = destination / file.path.generic_u8string().substr(prefix_length + 1);
             std::error_code ec;
-            const auto status = fs.symlink_status(file, ec);
+
+            fs.create_directory(target, ec);
             if (ec)
             {
-                msg::println_warning(format_filesystem_call_error(ec, "symlink_status", {file}));
-                continue;
+                msg::println_error(msgInstallFailed, msg::path = target, msg::error_msg = ec.message());
             }
+        }
 
-            const auto filename = file.filename();
-            if (vcpkg::is_regular_file(status) &&
-                (filename == "CONTROL" || filename == "vcpkg.json" || filename == "BUILD_INFO"))
+        // Copy files/symlinks
+        parallel_for_each(filtered_files, [&](const auto& file_and_status) {
+            auto& file = file_and_status.path;
+            auto& status = file_and_status.type;
+
+            if (is_directory(status)) return;
+
+            if (!is_regular_file(status) && !is_symlink(status))
             {
-                // Do not copy the control file or manifest file
-                continue;
+                Checks::unreachable(VCPKG_LINE_INFO);
+                return;
             }
 
-            const auto suffix = file.generic_u8string().substr(prefix_length + 1);
-            const auto target = destination / suffix;
+            const Path target = destination / file.generic_u8string().substr(prefix_length + 1);
+
+            if (fs.exists(target, IgnoreErrors{}))
+            {
+                msg::println_warning(msgOverwritingFile, msg::path = target);
+                if (is_regular_file(status))
+                {
+                    fs.remove(target, IgnoreErrors{});
+                }
+            }
 
             bool use_hard_link = true;
-            auto this_output = Strings::concat(destination_subdirectory, "/", suffix);
-            switch (status)
+            std::error_code ec;
+
+            if (is_regular_file(status))
             {
-                case FileType::directory:
+                if (use_hard_link)
                 {
-                    fs.create_directory(target, ec);
+                    fs.create_hard_link(file, target, ec);
                     if (ec)
                     {
-                        msg::println_error(msgInstallFailed, msg::path = target, msg::error_msg = ec.message());
+                        Debug::println("Install from packages to installed: Fallback to copy "
```

Comment:
`Debug::println` is not thread safe.

Reply:
There is other parallelized code where this wasn't a problem: in `cmd_execute_and_capture_output_parallel` we call `cmd_execute_and_stream_data` which calls `Debug::print()` (assuming you'd consider `Debug::print` also not thread safe).